src/gui/qgsmapcanvas.cpp

```diff
@@ -903,8 +902,19 @@ void QgsMapCanvas::showContextMenu( QgsMapMouseEvent *event )
 
   mMenu->addMenu( copyCoordinateMenu );
 
-  if ( mMapTool )
-    mMapTool->populateContextMenu( mMenu );
+}
+
+void QgsMapCanvas::showContextMenu( QgsMapMouseEvent *event )
+{
+
+  mMenu->clear();
+
+  // show map-tool actions first
+  if ( mMapTool ) mMapTool->populateContextMenu( mMenu );
```

Comment:
```suggestion
  if ( mMapTool )
    mMapTool->populateContextMenu( mMenu );
```

It's the coding style convention to line break even for short conditional statements.

Reply:
Thanks for your comment, I'll correct the code for this.
 
When I wrote ##37559, I never thought about that this might be of interest for the center QGIS map canvas as well. I just wanted to be able to modify the QgsMapCanvas context menu again. The update to QGIS 3.14 broke some of our python plugins ([EnMAP-Box](https://plugins.qgis.org/plugins/enmapboxplugin/) and [EO Time Series Viewer](https://plugins.qgis.org/plugins/timeseriesviewerplugin/)), which use inherited canvases e.g. to access the map layer properties via the context menu (example below).

![image](https://user-images.githubusercontent.com/1404870/86534070-d63f3700-bed5-11ea-886d-409f20e304b9.png)

Reply:
Thanks for the followup.

What I would propose instead of this PR is a minimal patch (as also suggested by @nyalldawson )

 - adding a new *signal*: `populateContextMenu( QMenu *menu, QgsMapMouseEvent *event )`
 - emitting this signal at the end of `showContextMenu()`

This way we have a single API for the main map canvas and inherited canvases.

Reply:
In that case I'd better name it ``contextMenuPopulated( QMenu *menu, QgsMapMouseEvent *event )``, for consistency with other signals.